File: src/modlunky2/sprites/journal_place.py
from pathlib import Path

from .base_classes import BaseSpriteLoader


class JournalPlaceSheet(BaseSpriteLoader):
    _sprite_sheet_path = Path("Data/Textures/journal_entry_place.png")
    _chunk_size = 320
    _chunk_map = {
        "journal_dwelling": (0, 0, 2, 1),
        "journal_jungle": (2, 0, 4, 1),
        "journal_volcana": (4, 0, 6, 1),
        "journal_olmecs_lair": (6, 0, 8, 1),
        "journal_tide_pool": (0, 1, 2, 2),
        "journal_temple_of_anubis": (2, 1, 4, 2),
        "journal_ice_caves": (4, 1, 6, 2),
        "journal_neo_babylon": (6, 1, 8, 2),
        "journal_sunken_city": (0, 2, 2, 3),
        "journal_cosmic_ocean": (2, 2, 4, 3),
        "journal_city_of_gold": (4, 2, 6, 3),
        "journal_duat": (6, 2, 8, 3),
        "journal_abzu": (0, 3, 2, 4),
        "journal_tiamats_throne": (2, 3, 4, 4),
        "journal_eggplant_world": (4, 3, 6, 4),
        "journal_hunduns_hideaway": (6, 3, 8, 4),
    }

File: crates/ml2_sprites/src/data/loaders/journal_place.rs
// Sprite loader/merger configs describing which chunks live in which
// sheet, keyed by tile name for the sprite fetcher lookup.

use crate::LoaderConfig;
use crate::data::chunks;

pub fn journal_place_sheet() -> LoaderConfig {
    LoaderConfig {
        name: "JournalPlaceSheet".into(),
        sprite_sheet_path: "Data/Textures/journal_entry_place.png".into(),
        chunk_size: 320,
        chunk_map: chunks(&[
            ("journal_dwelling", 0.0, 0.0, 2.0, 1.0),
            ("journal_jungle", 2.0, 0.0, 4.0, 1.0),
            ("journal_volcana", 4.0, 0.0, 6.0, 1.0),
            ("journal_olmecs_lair", 6.0, 0.0, 8.0, 1.0),
            ("journal_tide_pool", 0.0, 1.0, 2.0, 2.0),
            ("journal_temple_of_anubis", 2.0, 1.0, 4.0, 2.0),
            ("journal_ice_caves", 4.0, 1.0, 6.0, 2.0),
            ("journal_neo_babylon", 6.0, 1.0, 8.0, 2.0),
            ("journal_sunken_city", 0.0, 2.0, 2.0, 3.0),
            ("journal_cosmic_ocean", 2.0, 2.0, 4.0, 3.0),
            ("journal_city_of_gold", 4.0, 2.0, 6.0, 3.0),
            ("journal_duat", 6.0, 2.0, 8.0, 3.0),
            ("journal_abzu", 0.0, 3.0, 2.0, 4.0),
            ("journal_tiamats_throne", 2.0, 3.0, 4.0, 4.0),
            ("journal_eggplant_world", 4.0, 3.0, 6.0, 4.0),
            ("journal_hunduns_hideaway", 6.0, 3.0, 8.0, 4.0),
        ]),
    }
}
